lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -2589,6 +2589,204 @@ class ConvertAtenGridSamplerOp : public OpConversionPattern<AtenGridSamplerOp> {
 };
 } // namespace
 
+namespace {
+class ConvertInterpolateOp
+    : public OpConversionPattern<Aten__InterpolateSizeListScaleListOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(Aten__InterpolateSizeListScaleListOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    std::string mode;
+    matchPattern(op.getMode(), m_TorchConstantStr(mode));
+    if (mode != "bilinear") {
+      return failure();
+    }
+
+    Location loc = op->getLoc();
+    Value input = adaptor.getInput();
+    auto inputType = input.getType().cast<RankedTensorType>();
+    auto inputRank = inputType.getRank();
+
+    auto resultType = getTypeConverter()
+                          ->convertType(op.getResult().getType())
+                          .cast<RankedTensorType>();
+
+    SmallVector<Value, 2> outputSizeTorchInt;
+    if (!getListConstructElements(op.getSize(), outputSizeTorchInt))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: the output_size is not constructed from "
+              "ListConstruct");
+    SmallVector<Value, 2> outputSizeIntValues;
+    outputSizeIntValues = getTypeConvertedValues(
+        rewriter, loc, getTypeConverter(), outputSizeTorchInt);
+
+    Value outTensor = rewriter.create<tensor::EmptyOp>(
+        loc, resultType.getShape(), inputType.getElementType());
+
+    AffineMap idMap = rewriter.getMultiDimIdentityMap(inputRank);
+
+    SmallVector<utils::IteratorType> iteratorTypes(
+        inputRank, utils::IteratorType::parallel);
+
+    Value finalRes =
+        rewriter
+            .create<linalg::GenericOp>(
+                loc, outTensor.getType(), ValueRange{}, outTensor,
+                /*indexingMaps=*/idMap,
+                /*iteratorTypes=*/iteratorTypes,
+                [&](OpBuilder &b, Location loc, ValueRange args) {
```

Comment:
Given the length it is better to not use the body builder and do something similar to the following:

https://github.com/openxla/iree/blob/main/compiler/plugins/input/StableHLO/Conversion/StableHLOToLinalg.cpp#L2509

This avoids the multiple identations.